### PR TITLE
[SPARK-18973][SQL] Remove SortPartitions and RedistributeData

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?
SortPartitions and RedistributeData logical operators are not actually used and can be removed. Note that we do have a Sort operator (with global flag false) that subsumed SortPartitions.

## How was this patch tested?
Also updated test cases to reflect the removal.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/partitioning.scala [hunk 1]
```diff
@@ -1,49 +0,0 @@
...
-
-/**
- * This method repartitions data using [[Expression]]s into `numPartitions`, and receives
- * information about the number of partitions during execution. Used when a specific ordering or
- * distribution is expected by the consumer of the query result. Use [[Repartition]] for RDD-like
- * `coalesce` and `repartition`.
- * If `numPartitions` is not specified, the number of partitions will be the number set by
- * `spark.sql.shuffle.partitions`.
- */
-case class RepartitionByExpression(
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/partitioning.scala]: This is moved into basicLogicalOperators.scala

**Comment by rxin** [comment]
Please also merge this into branch-2.1 to minimize backport conflicts ...

**Comment by rxin** [comment]
Note that this is code from the initial Spark SQL commit!

**Comment by hvanhovell** [comment]
LGTM. Merging to master/2.1. Thanks!

**Pull request closed without merging**